model/jpa/src/main/java/org/jboss/aerogear/unifiedpush/jpa/dao/impl/JPAPushMessageInformationDao.java

```diff
@@ -141,6 +141,13 @@ public long getNumberOfPushMessagesForApplications() {
                 "IN (select p.pushApplicationID from PushApplication p)", Long.class).getSingleResult();
     }
 
+    @Override
+    public PushMessageInformation getPushMessageInformation(String id) {
+        return getSingleResultForQuery(createQuery(
+                "select pmi from PushMessageInformation pmi where pmi.id = :id")
+                .setParameter("id", id));
+
```

Comment:
@sebastienblanc if that is the PK, why not using `em.find(PushMessageInformation.class, id);` ?